nifi-nar-bundles/nifi-dropbox-bundle/nifi-dropbox-processors/src/main/java/org/apache/nifi/processors/dropbox/FetchDropbox.java

```diff
@@ -0,0 +1,179 @@
+            new Relationship.Builder().name("failure")
+                    .description("A FlowFile will be routed here for each File for which fetch was attempted but failed.")
+                    .build();
+    public static final Set<Relationship> relationships = Collections.unmodifiableSet(new HashSet<>(Arrays.asList(
+            REL_SUCCESS,
+            REL_FAILURE
+    )));
+
+    private static final List<PropertyDescriptor> PROPERTIES = Collections.unmodifiableList(Arrays.asList(
+            CREDENTIAL_SERVICE,
+            FILE,
+            ProxyConfiguration.createProxyConfigPropertyDescriptor(false, ProxySpec.HTTP_AUTH)
+    ));
+
+    private DbxClientV2 dropboxApiClient;
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        final ProxyConfiguration proxyConfiguration = ProxyConfiguration.getConfiguration(context);
+        dropboxApiClient = getDropboxApiClient(context, proxyConfiguration);
+    }
+
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        String fileIdentifier = context.getProperty(FILE).evaluateAttributeExpressions(flowFile).getValue();
+        fileIdentifier = correctFilePath(fileIdentifier);
+
+        FlowFile outFlowFile = flowFile;
+        try {
+            fetchFile(fileIdentifier, session, outFlowFile);
+            session.transfer(outFlowFile, REL_SUCCESS);
+        } catch (Exception e) {
+            handleError(session, flowFile, fileIdentifier, e);
+        }
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return PROPERTIES;
+    }
+
+    protected DbxClientV2 getDropboxApiClient(ProcessContext context, ProxyConfiguration proxyConfiguration) {
+        Proxy proxy = proxyConfiguration.createProxy();
+        StandardHttpRequestor.Config requestorConfig = StandardHttpRequestor.Config.builder()
+                .withProxy(proxy)
+                .build();
+        StandardHttpRequestor httpRequestor = new StandardHttpRequestor(requestorConfig);
+        DbxRequestConfig config = DbxRequestConfig.newBuilder(format("%s-%s", getClass().getSimpleName(), getIdentifier()))
+                .withHttpRequestor(httpRequestor)
+                .build();
+
+        final DropboxCredentialService credentialService = context.getProperty(CREDENTIAL_SERVICE)
+                .asControllerService(DropboxCredentialService.class);
+        DropboxCredentialDetails credential = credentialService.getDropboxCredential();
+
+        return new DbxClientV2(config, new DbxCredential(credential.getAccessToken(), -1L,
+                credential.getRefreshToken(), credential.getAppKey(), credential.getAppSecret()));
+    }
```

Comment:
@krisztina-zsihovszki I checked the code changes at high level and this method is common with `ListDropBox`. Also the `CREDENTIAL_SERVICE` property. They can be extracted into an abstract class or similar place.